modules/codegen/src/smithy4s/codegen/Codegen.scala

```diff
@@ -46,7 +46,9 @@ object Codegen { self =>
         scalaFile
       }
       val generatedNamespaces = codegenResult.map(_._2.namespace).distinct
-      val resources = if (!args.skipResources) {
+      val skipResource =
+        args.skipResources || (args.specs.isEmpty && generatedNamespaces.isEmpty)
```

Comment:
Why do we need that second part?

Reply:
☝️ I'm curious too. In particular, `generatedNamespaces` means "the namespaces that are already generated", not "the namespaces we're intending to generate"

Reply:
otherwise we produce two files:
```
> smithy4s generate
/Users/David.Francoeur/workspace/dev/smithy4s/META-INF/smithy/manifest
/Users/David.Francoeur/workspace/dev/smithy4s/META-INF/smithy/smithy4s.tracking.smithy

> cat META-INF/smithy/manifest
smithy4s.tracking.smithy
> cat META-INF/smithy/smithy4s.tracking.smithy 
$version: "2.0"

metadata smithy4sGenerated = [{smithy4sVersion: "dev-SNAPSHOT", namespaces: []}]
```
So my warning never prints

Reply:
ooh, gotcha

Reply:
smort

Reply:
normally I'd like that to be in a comment as it's more findable than a PR comment, but I don't want to block this - feel free to follow up or add it in another PR